core/src/SYCL/Kokkos_SYCL_Instance.hpp

```diff
@@ -330,8 +330,8 @@ struct sycl::is_device_copyable<
     Kokkos::Experimental::Impl::SYCLFunctionWrapper<Functor, Storage, false>>
     : std::true_type {};
 
-// FIXME_SYCL Remove when this specialization when specializations for
-// sycl::device_copyable also apply to const-qualified types.
+#if defined(KOKKOS_ENABLE_SYCL) && defined(__INTEL_LLVM_COMPILER) && \
+    __INTEL_LLVM_COMPILER < 20240000
```

Comment:
Better macros to check would be `__LIBSYCL_MAJOR_VERSION` and `__LIBSYCL_MINOR_VERSION`. They could be used to distinguish between release versions and would work with current development trunk. Won't work for arbitrary commits between two adjacent releases though, but nothing would to the best of my knowledge.

Reply:
How do I know which commit on `intel/llvm` corresponds to which `__LIBSYCL_MAJOR_VERSION/__LIBSYCL_MINOR_VERSION`?

Reply:
It's only meaningful for releases and the latest commit in intel/llvm, so I don't think the question is valid. I'd argue that there should be no expectations to support old commits of intel/llvm outside official releases (neither for us at intel/llvm nor for any library like kokkos).

Reply:
Let me rephrase my questions: What is the mapping from compiler releases to `__LIBSYCL_MAJOR_VERSION/__LIBSYCL_MINOR_VERSION`? I am also having a hard time mapping `intel/llvm` releases to `oneAPI` releases (and the other way around).

Reply:
Current intel/llvm trunk: 7/1, will be the same for OneAPI 2024.1
OneAPI 2024.0 7/0

Reply:
Hmmm... It's arguably easier to understand `__INTEL_LLVM_COMPILER < 20240000` than `_LIBSYCL_MAJOR_VERSION < 7`.

Reply:
I think the former is specific to OneAPI releases and not available for github.com/intel/llvm compilers.

Reply:
I added a commit to also check for that macro.